api-flask/app/caching.py

```diff
@@ -24,16 +28,33 @@ def cache_file(url, prefix, extension='cache'):
     # If the file exists, return path.
     if os.path.isfile(cache_filepath):
         logger.info('Returning cached file for {}.'.format(url))
-        return cache_filepath
+
+        # if the file is a geotiff, confirm that we can open it.
+        if extension == 'tif':
+            try:
+                rasterio.open(cache_filepath)
+                return cache_filepath
+            except Exception:
```

Comment:
We would need to specify the type of exception we are silencing. For this case, we could wrap all possible errors from rasterio with **rasterio.errors.RasterioError** or *rasterio.errors.RasterioIOError*